spaceship/lib/spaceship/connect_api/models/device.rb

```diff
@@ -54,8 +54,26 @@ def self.all(client: nil, filter: {}, includes: nil, limit: nil, sort: nil)
         return resps.flat_map(&:to_models)
       end
 
+      # @param client [ConnectAPI] ConnectAPI client.
+      # @param platform [String] The platform of the device.
+      # @param include_disabled [Bool] Whether to include disable devices. false by default.
+      # @return (Device) Find a device based on the UDID of the device. nil if no device was found.
+      def self.find_by_udid(device_udid, client: nil, platform: nil, include_disabled: false)
+        self.all(client: client).find do |device|
+          device.udid.casecmp(device_udid) == 0 && (include_disabled ? true : device.enabled?)
+        end
+      end
+
       def self.create(client: nil, name: nil, platform: nil, udid: nil)
         client ||= Spaceship::ConnectAPI
+        
+        # Find the device by UDID and return it in case it already exists
+        existing = self.find_by_udid(udid, client: client, platform: platform)
+        if existing
+          UI.verbose("UDID '#{udid}' already exists - Skipping its creation...")
```

Comment:
This line uses `UI`, which is not accessible without importing the module.

1) Idk which module to import, or how 😅 need advise here 😬 
2) This leads me to think that a model class shouldn't have UI-related logic probably... 🤔 however I wouldn't like to give up on this handy log. Not sure how to proceed here, as I'm not familiar with architecture in ruby projects 😅 

Any help is appreciated! Right now `rubocop` is failing because of this line

Reply:
Can you use `puts` instead? I think my experience level is similar to yours in this domain though :)

Reply:
I could yeah, but that'd mean it would always be printed, as opposed to using the `UI` logging utilities 😅 logging utilities are great to filter out noisy information, and also provide nice coloring to the output logs

Reply:
I'm removing it
```suggestion
```